apps/chat/src/components/Chat/Publish/ApproveRequiredSection.tsx

```diff
@@ -0,0 +1,178 @@
+import { IconClipboard } from '@tabler/icons-react';
+import { useEffect, useState } from 'react';
+
+import { useTranslation } from 'next-i18next';
+
+import classNames from 'classnames';
+
+import { getPublicationId } from '@/src/utils/app/publications';
+
+import { BackendResourceType } from '@/src/types/common';
+import { FolderSectionProps } from '@/src/types/folder';
+import { Publication, PublicationInfo } from '@/src/types/publication';
+import { Translation } from '@/src/types/translation';
+
+import {
+  ConversationsActions,
+  ConversationsSelectors,
+} from '@/src/store/conversations/conversations.reducers';
+import { useAppDispatch, useAppSelector } from '@/src/store/hooks';
+import {
+  PublicationActions,
+  PublicationSelectors,
+} from '@/src/store/publication/publication.reducers';
+
+import CaretIconComponent from '../../Common/CaretIconComponent';
+import CollapsibleSection from '../../Common/CollapsibleSection';
+import {
+  ConversationPublicationResources,
+  PromptPublicationResources,
+} from './PublicationResources';
+
+import { some } from 'lodash-es';
+
+interface PublicationProps {
+  publication: PublicationInfo & Partial<Publication>;
+  resourceType: BackendResourceType;
+}
+
+const PublicationItem = ({ publication, resourceType }: PublicationProps) => {
+  const dispatch = useAppDispatch();
+
+  const selectedPublication = useAppSelector(
+    PublicationSelectors.selectSelectedPublication,
+  );
+  const selectedConversationIds = useAppSelector(
+    ConversationsSelectors.selectSelectedConversationsIds,
+  );
+
+  const [isOpen, setIsOpen] = useState(false);
+
+  const ResourcesComponent =
+    resourceType === BackendResourceType.CONVERSATION
+      ? ConversationPublicationResources
+      : PromptPublicationResources;
+
+  return (
+    <div className="flex flex-col gap-1">
+      <div
+        onClick={() => {
+          setIsOpen(true);
+          if (!isOpen) {
+            dispatch(
+              PublicationActions.uploadPublication({ url: publication.url }),
+            );
+          } else {
+            dispatch(
+              PublicationActions.selectPublication({
+                publication: publication as Publication,
+              }),
+            );
+          }
+
+          dispatch(
+            ConversationsActions.selectConversations({
+              conversationIds: [],
+            }),
+          );
+        }}
+        className={classNames(
+          'group relative flex h-[30px] items-center rounded border-l-2 hover:bg-accent-primary-alpha',
+          selectedPublication?.url === publication.url &&
```

Comment:
Are you sure that we need to use property `url` instead of `id` here and in other places?
I believe we can everywhere use `id` for consistency with other entities (maybe interface `Entity` with `status`)

Reply:
publications do not have id, they have url. we could convert the data, like url to id, but I'm not sure that's a good idea